packages/forms/src/UIForm/UIForm.container.test.js

```diff
@@ -12,273 +15,381 @@ describe('UIForm container', () => {
 
 	it('should render form', () => {
 		// when
-		const wrapper = shallow(<UIForm data={data} {...props} />);
+		render(<UIForm data={getMockData()} {...props} />);
 
 		// then
 		expect(
-			wrapper // eslint-disable-line no-underscore-dangle
-				.instance().state,
-		).toMatchSnapshot();
-		expect(wrapper.getElement()).toMatchSnapshot();
+			screen.getByRole('textbox', { name: 'Last Name (with description)' }),
+		).toBeInTheDocument();
+		expect(
+			screen.getByRole('textbox', { name: 'First Name (with placeholder)' }),
+		).toBeInTheDocument();
+		expect(screen.getByRole('button', { name: 'Check the thing' })).toBeInTheDocument();
+		expect(screen.getByRole('button', { name: 'Submit' })).toBeInTheDocument();
 	});
 
-	describe('#UNSAFE_componentWillReceiveProps', () => {
-		it('should update state if form data structure changed', () => {
-			// given
-			const wrapper = shallow(<UIForm data={data} {...props} />);
+	it('should render a custom tag', () => {
+		// when
+		const dom = render(
+			<form>
+				<UIForm data={getMockData()} {...props} as="div" />
+			</form>,
+		);
 
-			const previousState = wrapper.state();
+		// then
+		// eslint-disable-next-line testing-library/no-container
+		const customElementForm = dom.container.querySelector('div.tf-uiform');
+		expect(customElementForm).toBeDefined();
```

Comment:
Or you can test a more realistic situation where the form is rendered as a fieldset that has an implicit role group